src/material/badge/badge.ts

```diff
@@ -238,15 +254,37 @@ export class MatBadge extends _MatBadgeBase implements OnInit, OnDestroy, CanDis
       this._badgeElement.textContent = newContentNormalized;
     }
 
+    if (!this._isHostInteractive()) {
+      if (!this._inlineBadgeDescription) {
+        this._inlineBadgeDescription = this._renderer.createElement('span') as HTMLSpanElement;
+        this._inlineBadgeDescription.classList.add('cdk-visually-hidden');
+      }
+      this._inlineBadgeDescription.textContent = this.description;
+
+      const host = this._elementRef.nativeElement;
+      host.parentElement?.insertBefore(this._inlineBadgeDescription, host.nextSibling);
```

Comment:
Does this have to be a sibling of the badge? Maybe we can add the element inside the badge?